### PR TITLE
Update user brand

### PR DESCRIPTION
**Pull request by jchen293**
# What
Added support for updating the user brand in the Golang library, updated the changelog, and fixed a typo.

# Why
This is one of the missing core features.

# How
Added methods for updating the user brand and add a new file for a code example. Add a new class for Brand.

# Testing
E2E local testing, no unit test was written because there is my personal information in the cassette.

### CODE REVIEW

**Review by nwithan8 — APPROVED** [review]

File: brand.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package easypost
+
+type Brand struct {
```
Comment (Justintime50) [inline brand.go:3]: Please add a comment above this line explaining what a brand is like all other structs have

**Review by Justintime50 — CHANGES_REQUESTED** [review]
Code looks good, let's just add a struct comment and then we can merge.

File: user.go [hunk 2]
```diff
@@ -113,3 +116,20 @@ func (c *Client) RetrieveMeWithContext(ctx context.Context) (out *User, err erro
 	c.get(ctx, "users", &out)
 	return
 }
+
+// UpdateBrand updates the user brand.
+func (c *Client) UpdateBrand(params map[string]interface{}, userID string) (out *Brand, err error) {
+	newParams := map[string]interface{}{"brand": params}
+	updateBrandUrl := fmt.Sprintf("users/%s/brand", userID)
```
Comment (Justintime50) [inline user.go:123]: Let's update `updateBrandUrl` to `updateBrandURL` to keep the linter happy.

**Review by Justintime50 — CHANGES_REQUESTED** [review]

File: user.go [hunk 3]
```diff
@@ -113,3 +116,20 @@ func (c *Client) RetrieveMeWithContext(ctx context.Context) (out *User, err erro
...
+
+// UpdateBrand updates the user brand.
+func (c *Client) UpdateBrand(params map[string]interface{}, userID string) (out *Brand, err error) {
+	newParams := map[string]interface{}{"brand": params}
+	updateBrandURL := fmt.Sprintf("users/%s/brand", userID)
+	c.put(nil, updateBrandURL, newParams, &out)
+	return
+}
+
+// UpdateBrand performs the same operation as UpdateBrand, but allows
```
Comment (Justintime50) [inline user.go:128]: Go docs expect the first word of the doc to be the function name, so here: `UpdateBrand` should become `UpdateBrandWithContext`.

**Review by Justintime50 — CHANGES_REQUESTED** [review]

File: user.go (hunk 3 above)
Comment (nwithan8) [reply user.go]: Wow, Godocs are really underwhelming for a modern language: https://go.dev/blog/godoc

**Review by Justintime50 — APPROVED** [review]

**Pull request merged**